lib/org/chromium/webidl/FetchSpecRegistrySelector.js

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+foam.CLASS({
+  package: 'org.chromium.webidl',
+  name: 'FetchSpecRegistrySelector',
+  implements: [
+    'foam.box.RegistrySelector',
+    'foam.core.StubFactory',
+  ],
+
+  requires: [
+    'foam.box.BoxRegistry',
+    'foam.box.Context',
+    'foam.box.node.ForkBox',
+    'foam.core.StubFactorySingleton',
+  ],
+  imports: ['registry'],
+
+  properties: [
+    {
+      name: 'defaultRegistry',
+      documentation: 'Registry for everything other than FetchSpecRunners.',
+      factory: function() { return this.Context.create().registry; }
+    },
+    {
+      name: 'fetchRegistry',
+      documentation: 'Registry used for FetchSpecRunners.',
+      factory: function() {
+        // Use defaultRegistry as context to ensure a common foam.box.Context
+        // namespace for all pipeline steps.
+        return this.get(this.BoxRegistry).create({
+          delegate: this.ForkBox.create({
+            childScriptPath: require('path')
+                .resolve(`${__dirname}/../../../../main/forkScript.js`)
+          }, this.defaultRegistry),
+        }, this.defaultRegistry);
+      },
+    },
+  ],
+
+  methods: [
+    function select(name, service, box) {
+      if (box && box.data && box.data.cls_ &&
```

Comment:
I was definitely thrown off guard when I was adding a test for this. If we decide to stick with this 'reaching into implementation details' thing, it might be a good idea to add a check that verifies that the box provided is a `SkeletonBox`. It was not immediately obvious from reading the code alone, and this might result in misuse of the selector.